.github/workflows/ci.yml

```diff
@@ -14,17 +14,18 @@ jobs:
       matrix:
         os: ['ubuntu-latest', 'windows-latest']
 
-    name: Node 16 on ${{ matrix.os }}
+    name: Node on ${{ matrix.os }}
     runs-on: ${{ matrix.os }}
 
     timeout-minutes: 30
     steps:
-      - uses: actions/setup-node@v1
+      - name: Checkout Code
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # 3.5.2
+      - uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # 3.6.0
         with:
-          node-version: 16.x
-      - uses: actions/checkout@v1
+          node-version-file: './package.json'
       - name: Restore Dependency Cache
-        uses: actions/cache@v3
+        uses: actions/cache@2b250bc32ad02700b996b496c14ac8c2840a2991
```

Comment:
Based on https://github.com/actions/cache/releases, this appears to be for a patch release of 2.18.x. Could we `88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8` use instead?
```suggestion
        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # 3.3.1
```
![Screenshot 2023-04-26 at 4 45 15 PM](https://user-images.githubusercontent.com/1930213/234698405-10d05e9f-300f-4ee0-b0ea-dd7ef9db6625.png)

Reply:
Nice. Thanks for the catch.